x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx

```diff
@@ -395,12 +398,15 @@ export const sendAlertToTimelineAction = async ({
   const ruleNote = getField(ecsData, ALERT_RULE_NOTE);
   const noteContent = Array.isArray(ruleNote) && ruleNote.length > 0 ? ruleNote[0] : '';
   const ruleTimelineId = getField(ecsData, ALERT_RULE_TIMELINE_ID);
-  const timelineId =
-    Array.isArray(ruleTimelineId) && ruleTimelineId.length > 0 ? ruleTimelineId[0] : '';
+  const timelineId = !isEmpty(ruleTimelineId)
+    ? Array.isArray(ruleTimelineId)
+      ? ruleTimelineId[0]
+      : ruleTimelineId
+    : '';
   const { to, from } = determineToAndFrom({ ecs });
 
   // For now we do not want to populate the template timeline if we have alertIds
-  if (!isEmpty(timelineId) && isEmpty(alertIds)) {
+  if (!isEmpty(timelineId) && isThresholdRule(ecsData) === false) {
```

Comment:
to ensure the template is applied to threshold rules, consider the following change suggested by @michaelolo24 :

```tsx
if (!isEmpty(timelineId) && !isEmpty(alertIds)) {
```